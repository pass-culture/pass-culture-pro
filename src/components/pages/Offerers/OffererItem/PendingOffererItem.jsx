import React from 'react'

import { formatSirenOrSiret } from '../../../../utils/siren'

const PendingOffererItem = ({ offerer }) => (
  <li className="offerer-item pending">
    <div className="list-content">
      <p>
        <span className="name">
          {offerer.name}
        </span>
        {` (SIREN: ${formatSirenOrSiret(offerer.siren)})`}
      </p>
      <p
        className="is-italic mb12"
        id="offerer-item-validation"
      >
        {'Rattachement en cours de validation'}
      </p>
    </div>
  </li>
)

export default PendingOffererItem
