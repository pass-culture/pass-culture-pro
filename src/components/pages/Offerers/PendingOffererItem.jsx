import React from 'react'
import { formatSire } from 'utils/sire'

const PendingOffererItem = ({ offerer }) => (
  <li className="offerer-item pending">
    <div className="list-content">
      <p>
        <span className="name">{offerer.name}</span> (SIREN:{' '}
        {formatSire(offerer.siren)})
      </p>
      <p className="is-italic mb12" id="offerer-item-validation">
        Rattachement en cours de validation par la structure
      </p>
    </div>
  </li>
)

export default PendingOffererItem
