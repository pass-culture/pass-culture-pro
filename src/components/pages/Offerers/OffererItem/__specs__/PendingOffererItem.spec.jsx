import React from 'react'
import { shallow } from 'enzyme'

import PendingOffererItem from '../PendingOffererItem'

describe('src | components | pages | Offerers | OffererItem | PendingOffererItem', () => {
  let props

  beforeEach(() => {
    props = {
      offerer: {}
    }
  })

  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<PendingOffererItem {...props} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
