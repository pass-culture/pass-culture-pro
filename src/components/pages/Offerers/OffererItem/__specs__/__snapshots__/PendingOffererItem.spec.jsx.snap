// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offerers | OffererItem | PendingOffererItem snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PendingOffererItem
    offerer={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="list-content"
      >
        <p>
          <span
            className="name"
          />
           (SIREN:
           
          
          )
        </p>
        <p
          className="is-italic mb12"
          id="offerer-item-validation"
        >
          Rattachement en cours de validation
        </p>
      </div>,
      "className": "offerer-item pending",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            <span
              className="name"
            />
             (SIREN:
             
            
            )
          </p>,
          <p
            className="is-italic mb12"
            id="offerer-item-validation"
          >
            Rattachement en cours de validation
          </p>,
        ],
        "className": "list-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="name"
              />,
              " (SIREN:",
              " ",
              "",
              ")",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "name",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            " (SIREN:",
            " ",
            "",
            ")",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rattachement en cours de validation",
            "className": "is-italic mb12",
            "id": "offerer-item-validation",
          },
          "ref": null,
          "rendered": "Rattachement en cours de validation",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="list-content"
        >
          <p>
            <span
              className="name"
            />
             (SIREN:
             
            
            )
          </p>
          <p
            className="is-italic mb12"
            id="offerer-item-validation"
          >
            Rattachement en cours de validation
          </p>
        </div>,
        "className": "offerer-item pending",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <span
                className="name"
              />
               (SIREN:
               
              
              )
            </p>,
            <p
              className="is-italic mb12"
              id="offerer-item-validation"
            >
              Rattachement en cours de validation
            </p>,
          ],
          "className": "list-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="name"
                />,
                " (SIREN:",
                " ",
                "",
                ")",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "name",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              " (SIREN:",
              " ",
              "",
              ")",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rattachement en cours de validation",
              "className": "is-italic mb12",
              "id": "offerer-item-validation",
            },
            "ref": null,
            "rendered": "Rattachement en cours de validation",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
